tests/specs/builds/symlink.ts

```diff
@@ -0,0 +1,127 @@
+import { testSuite, expect } from 'manten';
+import { createFixture } from 'fs-fixture';
+import { pkgroll } from '../../utils.js';
+import {
+	installTypeScript,
+	createPackageJson,
+	createTsconfigJson,
+} from '../../fixtures.js';
+
+export default testSuite(({ describe }, nodePath: string) => {
+	describe('symlink', ({ test }) => {
+		test('pnpm monorepo', async () => {
+			await using fixture = await createFixture({
+				...installTypeScript,
+				packages: {
+					one: {
+						'package.json': createPackageJson({
+							type: 'module',
+							main: './dist/index.cjs',
+							module: './dist/index.mjs',
+							types: './dist/index.d.cts',
+							dependencies: {
+								vue: '*',
+							},
+							peerDependencies: {
+								vue: '*',
+							},
+						}),
+						'tsconfig.json': createTsconfigJson({
+							extends: '../../tsconfig.base.json',
+							compilerOptions: {
+								preserveSymlinks: false,
+							},
+							include: ['src'],
+							exclude: ['**/dist', '**/node_modules'],
+						}),
+						node_modules: {
+							vue: {
+								'index.d.ts': ({ symlink }) => symlink('../../../../node_modules/.pnpm/vue/index.d.ts'),
+								node_modules: {
+									'@vue': {
+										'runtime-dom': ({ symlink }) => symlink('../../../../node_modules/.pnpm/@vue/runtime-dom'),
+									},
+								},
+							},
+						},
+						'src/index.ts': `
+						import { defineComponent, h } from 'vue';
+						export default defineComponent({
+              props: {
+                message: String,
+              },
+              setup(props) {
+                h('div', {
+                  ...props,
+                });
+              },
+            });
+						`,
+					},
+				},
+				'tsconfig.base.json': createTsconfigJson({
+					compilerOptions: {
+						module: 'ESNext',
+						target: 'ES2020',
+						lib: ['DOM', 'DOM.Iterable', 'ES2020'],
+						moduleResolution: 'Bundler',
+						strict: true,
+					},
+				}),
+				'package.json': createPackageJson({
+					type: 'module',
+				}),
+				'node_modules/': {
+					'.pnpm': {
+						'@vue/runtime-dom/index.d.ts': `
+              type ComponentProps = {
+                props: {
+                  message: StringConstructor;
+                };
+                setup: (props: ComponentProps['props']) => any;
+              };
+              export declare function h(el: string, props: ComponentProps['props']): any;
+              export declare function defineComponent({
+                props,
+                setup,
+              }: ComponentProps): ComponentProps;
+              `,
+						vue: {
```

Comment:
Fixtures can be a lot simpler if we keep them pure and simple. Instead of trying to mock something thats a lot more than this, we should just use `dep-a` and isolate the test to just testing symlinks.